### PR TITLE
Ability to work without sprockets

### PR DESCRIPTION
**Pull request by fc-arny**
Sprockets'  tasks (assets:*)  not available if you do not require 'sprockets/railtie' and when you try run some rake task got a message: 
```bash
ReactOnRails: Set generated_assets_dir to default: app/assets/webpack
Running via Spring preloader in process 18586
rake aborted!
Don't know how to build task 'assets:precompile' (see --tasks)
/Users/arthur/.rvm/gems/ruby-2.3.3/gems/react_on_rails-6.3.4/lib/tasks/assets.rake:35:in `<top (required)>'
...
```
My application config

```ruby
# frozen_string_literal: true
require_relative 'boot'

require 'rails'

# Pick the frameworks you want:
require 'active_model/railtie'
require 'active_job/railtie'
require 'active_record/railtie'
require 'action_controller/railtie'
require 'action_mailer/railtie'
require 'action_view/railtie'
# require 'sprockets/railtie'

Bundler.require(*Rails.groups)

module My
  class Application < Rails::Application
    # Settings in config/environments/* take precedence over those specified here.
    # Application configuration should go into files in config/initializers
    # -- all .rb files in that directory are automatically loaded.

    config.i18n.load_path += Dir[Rails.root.join('config', 'locales', '**', '*.{rb,yml}').to_s]
    config.i18n.default_locale = :ru
    config.i18n.enforce_available_locales = true

    config.time_zone = 'Moscow'

    config.middleware.use I18n::JS::Middleware
  end
end
```

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/671)

### CODE REVIEW

**Comment by justin808** [comment]
@fc-arny can you explain the use case for not using sprockets?

If you're confident others can benefit from this change, please create a changelog entry and rebase your changes on top of master. https://github.com/shakacode/react_on_rails/blob/master/CONTRIBUTING.md

**Comment by fc-arny** [comment]
@justin808 my goal is use only one tool for compiling assets. 

Using webpack for compiling js, then pass bundle to application.js and then flow through sprockets' process  is very slow and not transparent. Now we use react_on_rails for component rendering and https://github.com/mipearson/webpack-rails(+ stats-webpack-plugin)  for inclusion code on page. 

I'm not sure we need put this case in changelog. I think we can implement some ideas from webpack-rails and create better completed solution later.

**Comment by justin808** [comment]
Sounds good. Please rebase your changes on top of master and get CI to pass. Please add a CHANGELOG.md entry and if possible, add some test.

**Comment by justin808** [comment]
@fc-arny Wow, this is really great. What I'd **LOVE** to see is a file in the docs/additional-reading directory on how you use the github.com/mipearson/webpack-rails package with React on Rails.

How about we call the file: 

docs/additional-reading/rails-without-sprockets

Are you able to create a simple test deployment to Heroku with this setup?

Please provide as much as you can regarding what you learned and I'll contribute with you.

**Comment by fc-arny** [comment]
@justin808 It's really cool! Let's start with this simple approach 👍

**Comment by justin808** [comment]
@fc-arny Any progress on this one?

**Comment by fc-arny** [comment]
@justin808 in progress. Setup [sample application](https://github.com/atconsulting/no-sprockets-example)  and creating doc with this flow.

**Comment by justin808** [comment]
@fc-arny I'm going to push out a new release later today. If this is ready, I'll include it.

**Comment by justin808** [comment]
@fc-arny Anything I can to help with this one?

**Comment by fc-arny** [comment]
@justin808  sorry for this... just need some free time (3 days i this) and i will create sample application and documentation.

**Comment by justin808** [comment]
@fc-arny I'm looking forward to this one. Let me know if I can help. Please be sure to see https://github.com/shakacode/react_on_rails/blob/master/CONTRIBUTING.md for some details like adding tests, updating the CHANGELOG.md, etc.

**Comment by justin808** [comment]
@fc-arny Anything I can help you with to get this to completion?

**Comment by justin808** [comment]
@fc-arny @rupurt, I'd like to know which of the approaches to advocate

* This one #671 
* Rupert's: Possible Simplified Rails Configuration without Asset Pipeline #696

**Comment by justin808** [comment]
@fc-arny I just took a look at the code here. It seems safe enough. Just don't include the sprockets related code if Sprockets is not in the gemfile.

@rupurt, @robwise, @alexfedoseev Thoughts on this one?

**Comment by rupurt** [comment]
@justin808 I'd say you want a combination of both. It makes sense to guard the sprockets requires if the gem is not installed, but I would want to keep sprockets working for legacy apps so they can be gradually transitioned.

**Comment by justin808** [comment]
@fc-arny Let me know if this is ready for merge, or if you had some more docs that you wanted to add.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,8 @@ All notable changes to this project's source code will be documented in this fil
 Contributors: please follow the recommendations outlined at [keepachangelog.com](http://keepachangelog.com/). Please use the existing headings and styling as a guide, and add a link for the version diff at the bottom of the file. Also, please update the `Unreleased` link to compare to the latest release version.
 
 ## [Unreleased]
-*Please add entries here for your pull requests.*
```
Comment (justin808) [inline CHANGELOG.md]: @fc-arny Please don't remove this line. Let me know if this is all ready for merge otherwise.

**Comment by justin808** [comment]
@fc-arny Is this one ready? I'm going to do a release on Saturday.

**Comment by justin808** [comment]
@fc-arny Please update to master to fix the conflict.

**Comment by justin808** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/671#-:-KciyORLh79AgWIVnXe4:bnfp4nl)*

**Comment by justin808** [comment]
In 6.5.1.

**Pull request merged**